pkg/component/runtime/state.go

```diff
@@ -79,6 +84,8 @@ func newComponentState(comp *component.Component) (s ComponentState) {
 	s.Message = startingMsg
 	s.Units = make(map[ComponentUnitKey]ComponentUnitState)
 	s.expectedUnits = make(map[ComponentUnitKey]expectedUnitState)
+	s.expectedFeaturesIdx = 1
```

Comment:
I expected to see this initialized along with the config index. Why is this in a separate place?

Reply:
Moved to the same place where new units' `configStateIdx` is initialized - a0a24a59d.